src/components/certificates/EditCertificate.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import CertificateForm from './CertificateForm';
+import { CertificatesContext } from '../../contexts/certificatesContext';
+import { Certificate } from '../../types/types';
+
+const EditCertificate: React.FC = () => {
+  const { id } = useParams<{ id: string }>();
+  const { certificates, loading, error } = useContext(CertificatesContext)!;
+  const [certificate, setCertificate] = useState<Certificate | null>(null);
+
+  useEffect(() => {
+    if (certificates && id) {
+      const foundCertificate = certificates.find(
+        (cert: Certificate) => cert.id === parseInt(id, 10),
+      );
+      setCertificate(foundCertificate || null);
+    }
+  }, [certificates, id]);
```

Comment:
As expxlained above, you should fetch certificate by ID from DB (later from backend), not load here all certificates and search for the one